src/crypto/PSAOperationalKeystore.cpp

```diff
@@ -102,6 +113,9 @@ bool PSAOperationalKeystore::HasOpKeypairForFabric(FabricIndex fabricIndex) cons
 {
     VerifyOrReturnError(IsValidFabricIndex(fabricIndex), false);
 
+    // check if the key exists in the PersistentStorageOperationalKeystore exist, and if so move it to the PSA ITS secure storage.
+    VerifyOrReturnError(CHIP_NO_ERROR == MoveOpKeysFromKVSToITS(fabricIndex), false);
```

Comment:
On failure, this can leave the device unable to be reached on a given fabric.

All of the translation should be done *before* starting to process anything, not everytime HasOpKeypairForFabric is called.

I would recommend making a separate building block that a public method "MoveOpKeysFromPersistentStorageToITS" or similar name, that can be called *on purpose, by choice* by a given product, to proceed with the migration of all fabrics at once, when the FabricTable has been initialized, such as after Server Init but before processing any incoming connections. Perhaps there is a race with subscription resumption, and it has to be done with product knowledge of storage in a way that can be done outside Server::Init(), but the idea is the same: having this behavior by default for all adds a large amount of untested risk to all integrators who are making use of PSA, but had previously commissioned devices (e.g. inherit this new path via a firmware update).